Microsoft.Alm.Authentication/TargetUri.cs

```diff
@@ -112,22 +107,71 @@ public bool IsBaseOf(TargetUri targetUri)
             if (targetUri == null)
                 return false;
 
-            return ResolvedUri.IsBaseOf(targetUri.ResolvedUri);
+            return ActualUri.IsBaseOf(targetUri.ActualUri);
         }
         /// <summary>
-        /// Gets a canonical string representation for the <see cref="ResolvedUri"/>.
+        /// Gets a canonical string representation for the <see cref="ActualUri"/>.
         /// </summary>
         /// <returns></returns>
         public override String ToString()
         {
-            return ResolvedUri.ToString();
+            return ActualUri.ToString();
+        }
+
+        public HttpClientHandler HttpClientHandler
+        {
+            get
+            {
+                bool useProxy = ProxyUri != null;
+                return new HttpClientHandler()
+                {
+                    Proxy = WebProxy,
+                    UseProxy = useProxy,
+                    MaxAutomaticRedirections = 2,
+                    UseDefaultCredentials = !useProxy
+                };
+            }
+        }
+
+        public WebProxy WebProxy
+        {
+            get
+            {
+                if (ProxyUri != null)
+                {
+                    WebProxy proxy = new WebProxy(ProxyUri);
+
+                    int dividerIndex = ProxyUri.UserInfo.IndexOf(':');
+
+                    if (!string.IsNullOrWhiteSpace(ProxyUri.UserInfo)
+                        && dividerIndex != -1)
+                    {
+                        string userName = ProxyUri.UserInfo.Substring(0, dividerIndex);
+                        string password = ProxyUri.UserInfo.Substring(dividerIndex + 1);
+
+                        NetworkCredential proxyCreds = new NetworkCredential(
+                            userName,
+                            password
+                        );
+
+                        proxy.UseDefaultCredentials = false;
+                        proxy.Credentials = proxyCreds;
+                    }
+
```

Comment:
These are the four cases that we need to cater for. I think that you're good for cases 1, 2 & 4. If we can test 3, then it looks good to me.

<table>
    <tr>
        <th>#</th>
        <th>Case</th>
        <th>Code</th>
    </tr>
    <tr>
        <td>1</td>
        <td>No proxy required</td>
        <td>Nothing extra required</td>
    </tr>
    <tr>
        <td>2</td>
        <td>Proxy required, no proxy auth required</td>
        <td><pre>httpClientHandler.Proxy = new WebProxy(ProxyUri);
httpClientHandler.UseProxy = true;</pre></td>
    </tr>
    <tr>
        <td>3</td>
        <td>Proxy required, proxy auth with default credentials</td>
        <td><pre>httpClientHandler.Proxy = new WebProxy(ProxyUri);
httpClientHandler.UseProxy = true;
httpClientHandler.Proxy.Credentials = null;
httpClientHandler.Proxy.UseDefaultCredentials = true;</pre></td>
    </tr>
    <tr>
        <td>4</td>
        <td>Proxy required, proxy auth non-default credentials specified or required</td>
        <td><pre>httpClientHandler.Proxy = new WebProxy(ProxyUri);
httpClientHandler.UseProxy = true;
httpClientHandler.Proxy.Credentials = new NetworkCredential(proxyUsername, proxyPassword);
httpClientHandler.Proxy.UseDefaultCredentials = false;</pre></td>
    </tr>
</table>